lib/cisco_node_utils/bridge_domain_vni.rb

```diff
@@ -0,0 +1,165 @@
+    #                      PROPERTIES                      #
+    ########################################################
+
+    # Bridge-Domain member vni assigning case
+    # Not all the bds created or initialized in this class context can have the
+    # member vni's mapped. So will get the vnis of the ones which are mapped to
+    # the bds.
+    # Eg: Suppose the bd mapping is as below on the switch
+    # bridge-domain 100,105,107-109,150
+    #   member vni 5000, 8000, 5007-5008, 7000, 5050
+    # If puppet layer tries to get values of 100,105,107 bds as defined in
+    # manifest then the final_bd_vni map which is returned will contain only
+    # these mappings as below;
+    # '5000,8000,5007'
+    # This is also to handle the idempotence case, and if any mismatch then set
+    # the member_vni
+    def member_vni
+      bd_vni_hash = curr_bd_vni_hash
+      ret_list = []
+      @bd_ids_list.each do |bd|
+        ret_list << bd_vni_hash[bd]
+      end
+      Utils.unsorted_list_to_string(ret_list)
+    end
+
+    # This member_vni mapping will be executed only when the val is not empty
+    # else it will be treated as a 'no' cli and executed as required.
+    # If the mappings do not match in any fashion then cli normally returns a
+    # failure which will be handled.
+    # During set of member vni we first see if any bd is mapped to some other
+    # vni, if yes then remove that one and apply the new vni.
+    # Eg: bridge-domain 100-110
+    #       member vni 5100-5110
+    # Now the manifest is changed to 5100-5105,6106-6110, so hence first
+    # 5106-5110 mapping is removed and then 6106-6110 is applied.
+    def member_vni=(val)
+      val = val.to_s
+      Feature.vni_enable
+      bd_vni = curr_bd_vni_hash
+      if val.empty?
+        bd_val = bd_vni.keys.join(',')
+        vni_val = bd_vni.values.join(',')
+        return '' if vni_val.empty?
+        config_set('bridge_domain_vni', 'member_vni', vnistate: 'no',
+                   vni: vni_val, bd: bd_val, membstate: 'no', membvni: vni_val)
+      else
+        unless bd_vni.empty?
+          inp_vni_list = Utils.string_to_array(val.to_s)
+          inp_bd_vni_hash = Hash[@bd_ids_list.zip(inp_vni_list)]
+
+          temp_hash = bd_vni.to_a.keep_if { |k, _v| inp_bd_vni_hash.key? k }
+          rem_hash = (temp_hash.to_a - inp_bd_vni_hash.to_a).to_h
+
+          rm_bd = rem_hash.keys.join(',')
+          rm_vni = rem_hash.values.join(',')
+          config_set('bridge_domain_vni', 'member_vni', vnistate: 'no',
+                     vni: rm_vni, bd: rm_bd, membstate: 'no', membvni: rm_vni)
+        end
+        config_set('bridge_domain_vni', 'member_vni', vnistate: '', vni: val,
+                   bd: @bd_ids, membstate: '', membvni: val)
+      end
+    end
+
+    def default_member_vni
+      config_get_default('bridge_domain_vni', 'member_vni')
+    end
+
+    # getter for system bridge-domain
+    def system_bd_add
```

Comment:
Confusing name. I would just use `system_bridge_domain`. 

The `add` / `remove` keywords can be passed as variables if needed.